asyncio_mqtt/client.py

```diff
@@ -320,7 +320,7 @@ async def __aexit__(self, exc_type, exc, tb):
         # Early out if already disconnected...
         if self._disconnected.done():
             disc_exc = self._disconnected.exception()
-            if disc_exc is None:
+            if disc_exc is not None:
```

Comment:
This change isn't required to resolve the issue, but it seems like a typo so I included it. Let me know if you want that as a separate PR/discussion.